futaba/cogs/moderation/core.py

```diff
@@ -169,25 +176,28 @@ async def unmute(
         else:
             await self.bot.punish.unjail(ctx.guild, member, reason)
 
-    async def perform_jail(self, ctx, member, minutes, reason):
-        roles = self.bot.sql.settings.get_special_roles(ctx.guild)
+    async def perform_jail_internal(self, message, member, minutes, reason):
+        roles = self.bot.sql.settings.get_special_roles(message.guild)
         if roles.jail is None:
             raise CommandFailed(content="No configured jail role")
 
-        if member.top_role >= ctx.me.top_role:
+        if member.top_role >= self.bot.user.top_role:
```

Comment:
`User`s do not have roles. `ctx.me` is the `Member` version of the bot, which is associated with a guild and does have rules.

Reply:
Is there a way to get the ``Member`` version of the bot without using ``ctx`` since it would be impossible to use ``ctx`` in ``on_message``?